static/app/components/profiling/profilingFlamechartLayout.tsx

```diff
@@ -1,34 +1,83 @@
+import {
+  useResizableDrawer,
+  UseResizableDrawerOptions,
+} from 'sentry/utils/profiling/hooks/useResizableDrawer';
 
 interface ProfilingFlamechartLayoutProps {
-  flamechart: React.ReactNode;
-  frameStack: React.ReactNode;
-  minimap: React.ReactNode;
+  flamechart: React.ReactElement;
+  frameStack: React.ReactElement;
+  minimap: React.ReactElement;
 }
 
 export function ProfilingFlamechartLayout(props: ProfilingFlamechartLayoutProps) {
   const flamegraphTheme = useFlamegraphTheme();
   const {layout} = useFlamegraphPreferencesValue();
+  const frameStackRef = useRef<HTMLDivElement>(null);
+
+  const resizableOptions: UseResizableDrawerOptions = useMemo(() => {
+    const initialDimensions: [number, number] = [
+      // Half the screen minus the ~sidebar width
+      window.innerWidth * 0.5 - 220,
+      (flamegraphTheme.SIZES.FLAMEGRAPH_DEPTH_OFFSET + 2) *
+        flamegraphTheme.SIZES.BAR_HEIGHT,
+    ];
+
+    // 664 is approximately the width where we start to scroll inside
+    // 30 is the min height to where the drawer can still be resized
+    const min: [number, number] = [664, 30];
+
+    const onResize = (newDimensions: [number, number]) => {
+      if (!frameStackRef.current) {
+        return;
+      }
+
+      if (layout === 'table_left' || layout === 'table_right') {
+        frameStackRef.current.style.width = `${newDimensions[0]}px`;
+        frameStackRef.current.style.height = `100%`;
+      } else {
+        frameStackRef.current.style.height = `${newDimensions[1]}px`;
+        frameStackRef.current.style.width = `100%`;
+      }
+    };
+
+    return {
+      initialDimensions,
+      onResize,
+      direction:
+        layout === 'table_left'
+          ? 'horizontal-ltr'
+          : layout === 'table_right'
+          ? 'horizontal-rtl'
+          : 'vertical',
+      min,
+    };
+  }, [
+    flamegraphTheme.SIZES.FLAMEGRAPH_DEPTH_OFFSET,
+    flamegraphTheme.SIZES.BAR_HEIGHT,
+    layout,
+  ]);
+
+  const {onMouseDown} = useResizableDrawer(resizableOptions);
 
   return (
-    <Fragment>
-      <ProfilingFlamechartLayoutContainer>
-        <ProfilingFlamechartGrid layout={layout}>
-          <MinimapContainer height={flamegraphTheme.SIZES.MINIMAP_HEIGHT}>
-            {props.minimap}
-          </MinimapContainer>
-          <ZoomViewContainer>{props.flamechart}</ZoomViewContainer>
-          <FrameStackContainer layout={layout}>{props.frameStack}</FrameStackContainer>
-        </ProfilingFlamechartGrid>
-      </ProfilingFlamechartLayoutContainer>
-    </Fragment>
+    <ProfilingFlamechartLayoutContainer>
+      <ProfilingFlamechartGrid layout={layout}>
+        <MinimapContainer height={flamegraphTheme.SIZES.MINIMAP_HEIGHT}>
+          {props.minimap}
+        </MinimapContainer>
+        <ZoomViewContainer>{props.flamechart}</ZoomViewContainer>
+        <FrameStackContainer ref={frameStackRef} layout={layout}>
+          {cloneElement(props.frameStack, {onResize: onMouseDown})}
```

Comment:
I'm not a fan of cloning this, but I didnt want to create a context or move the framestack jsx down a level.